### PR TITLE
Fix web impl streaming line breaks with large rows

### LINKED ISSUE #205: Streaming line breaks in web version
See #171 & #204 

The web version may have the same issue.

### PR DESCRIPTION
**Pull request by slvrtrn**
## Summary

Resolves #205 
See also: #171, #204 (there was the same fix, but for Node.js impl only)

## Checklist
- [x] Unit and integration tests covering the common scenarios were added
- [x] A human-readable description of the changes was provided to include in CHANGELOG

### CODE REVIEW

**Comment by slvrtrn** [comment]
```
SUMMARY:
✔ 562 tests completed
ℹ 4 tests skipped
✖ 4 tests failed

FAILED TESTS:
  [Web] SELECT streaming
    should correctly process multiple chunks
      large amount of rows
        ✖ should work with .json()
          Chrome Headless 129.0.0.0 (Linux x86_64)
        TypeError: Failed to fetch
            at WebConnection.request (webpack/commons.js:8674:36)
            at WebConnection.insert (webpack/commons.js:8606:37)
            at WebClickHouseClientImpl.insert (webpack/commons.js:6159:46)
            at genLargeStringsDataset (webpack/commons.js:5526:18)
            at async UserContext.<anonymous> (webpack/commons.js:8252:43)

        ✖ should work with .stream()
          Chrome Headless 129.0.0.0 (Linux x86_64)
        TypeError: Failed to fetch
            at WebConnection.request (webpack/commons.js:8674:36)
            at WebConnection.insert (webpack/commons.js:8606:37)
            at WebClickHouseClientImpl.insert (webpack/commons.js:6159:46)
            at genLargeStringsDataset (webpack/commons.js:5526:18)
            at async UserContext.<anonymous> (webpack/commons.js:8265:43)

      rows that don't fit into a single chunk
        ✖ should work with .json()
          Chrome Headless 129.0.0.0 (Linux x86_64)
        TypeError: Failed to fetch
            at WebConnection.request (webpack/commons.js:8674:36)
            at WebConnection.insert (webpack/commons.js:8606:37)
            at WebClickHouseClientImpl.insert (webpack/commons.js:6159:46)
            at genLargeStringsDataset (webpack/commons.js:5526:18)
            at async UserContext.<anonymous> (webpack/commons.js:8281:43)

        ✖ should work with .stream()
          Chrome Headless 129.0.0.0 (Linux x86_64)
        TypeError: Failed to fetch
            at WebConnection.request (webpack/commons.js:8674:36)
            at WebConnection.insert (webpack/commons.js:8606:37)
            at WebClickHouseClientImpl.insert (webpack/commons.js:6159:46)
            at genLargeStringsDataset (webpack/commons.js:5526:18)
            at async UserContext.<anonymous> (webpack/commons.js:8294:43)
```

Apparently, POST with a very large body size does not work well with Chrome Runner :/
For now, it might be better to generate the test data on the CH side.

File: packages/client-web/src/result_set.ts [hunk 1]
```diff
@@ -67,41 +69,81 @@ export class ResultSet<Format extends DataFormat | unknown>
...
+            json<T>(): T {
+              return JSON.parse(text)
+            },
+          })
+          lastIdx = idx + 1 // skipping newline character
+
+          // done with the first row in the current chunk
+          // --
+          // maybe there are more rows in the current chunk
+          do {
```
Comment (mshustov) [inline packages/client-web/src/result_set.ts]: can it be merged with the logic above? the complexity of the function increased

**Review by mshustov — APPROVED** [review]

File: packages/client-web/src/result_set.ts (hunk 1 above)
Comment (slvrtrn) [reply packages/client-web/src/result_set.ts]: I guess the length check is cheap, so it's mergeable. 
Fixed.

Comment (slvrtrn) [reply packages/client-web/src/result_set.ts]: I will check if the Node.js impl can be simplified, too (there are also two separate passes with a similar logic), but that is for the follow-up.

**Pull request merged**